### PR TITLE
Adding autodetect of SH1106 or SSD1306 OLED displays 

### PR DESCRIPTION
**Pull request by mjstn**
We have a potential for production having to use the little OLED display on the MCB that may come with one or another OLED display controller chip.   

The OLED node prior to now just supported the SH1106 but did have code to allow support for SSD1306.

Being that it is a bit tricky doing detection of which chip we default to SH1106 which is what it 'should' be unless China messes up procurement.

I am asking you, Janez, to do this review because we as a team really have to spread the load for these reviews and this one is something that you may find interesting anyway.   So don't worry about the specifics of the chip itself but look for stupid errors and that is the MAIN point of a review is to improve code robustness.    Thanks.

### CODE REVIEW

File: src/oled_display_node.cpp
Comment (JanezCim) [inline src/oled_display_node.cpp]: Here and on a couple of other places you use printf instead of ROS_INFO, is there a reason for it?

File: src/oled_display_node.cpp
Comment (JanezCim) [inline src/oled_display_node.cpp:345]: I might be understanding wrongly but if the datasheet guarantees the lower 3 bits as 0 when the type is SH106, why wouldnt you rather check for that in the first if and then in the else assume that its a type SSD1306 if the if didnt pass? :)

**Review by JanezCim — COMMENTED** [review]
I have 2 questions, otherwise the code looks ok to me. But i have not compiled and run it, should that also be done?

File: src/oled_display_node.cpp
Comment (mjstn) [reply src/oled_display_node.cpp]: This is an oversight.   I will change this.  thank you

File: src/oled_display_node.cpp
Comment (mjstn) [reply src/oled_display_node.cpp:345]: Although that sounds reasonable., what I was going for is our 'master default' is the SH1106.   So if it did not match SSD1306  then no matter WHAT it was it would use SH1106.    i am being concerned that it is best to default what production SHOULD have bought so only deviate from that if I get a certain match it is the SSD1306 thus test only for SSD1306.    Sadly there is no 'chipid' register like most every other I2C chip I have used AND I have used quite a few.

File: src/oled_display_node.cpp
Comment (mjstn) [reply src/oled_display_node.cpp]: Await the push that resolves this prior to any comments on this printf issue is fine with me but your call.

**Comment by mjstn** [comment]
Requested changes have been pushed.   Also I exit node now if no display detected.  This will allow manual debug so the node is not active in an idle state.     Thanks for inputs.   Look at changes then approve if accepted.

File: src/oled_display_node.cpp
Comment (mjstn) [reply src/oled_display_node.cpp]: All the printf statements are removed so it's ready to approve

**Review by JanezCim — APPROVED** [review]

**Pull request merged**
